sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala

```diff
@@ -3102,27 +3102,77 @@ abstract class DDLSuite extends QueryTest with SQLTestUtils {
     }
   }
 
-  test("Move data to trash on truncate table if enabled") {
+  test("SPARK-32481 Move data to trash on truncate table if enabled") {
+    val trashIntervalKey = "fs.trash.interval"
     withTable("tab1") {
       withSQLConf(SQLConf.TRUNCATE_TRASH_ENABLED.key -> "true") {
         sql("CREATE TABLE tab1 (col INT) USING parquet")
         sql("INSERT INTO tab1 SELECT 1")
-
+        // scalastyle:off hadoopConfiguration
+        val hadoopConf = spark.sparkContext.hadoopConfiguration
+        // scalastyle:on hadoopConfiguration
+        val originalValue = hadoopConf.get(trashIntervalKey, "0")
         val tablePath = new Path(spark.sessionState.catalog
           .getTableMetadata(TableIdentifier("tab1")).storage.locationUri.get)
-        val hadoopConf = spark.sessionState.newHadoopConf()
-        val fs = tablePath.getFileSystem(hadoopConf)
-        // trash interval should be configured from hadoop side
-        hadoopConf.setInt("fs.trash.Interval", 5)
 
+        val fs = tablePath.getFileSystem(hadoopConf)
         val trashRoot = fs.getTrashRoot(tablePath)
         assert(!fs.exists(trashRoot))
-        sql("TRUNCATE TABLE tab1")
+        try {
+          hadoopConf.set(trashIntervalKey, "5")
+          sql("TRUNCATE TABLE tab1")
+        } finally {
+          hadoopConf.set(trashIntervalKey, originalValue)
+        }
         assert(fs.exists(trashRoot))
         fs.delete(trashRoot, true)
       }
     }
   }
+
+  test("SPARK-32481 delete data permanently on truncate table if trash interval is non-positive") {
+    val trashIntervalKey = "fs.trash.interval"
+    withTable("tab1") {
+      withSQLConf(SQLConf.TRUNCATE_TRASH_ENABLED.key -> "true") {
+        sql("CREATE TABLE tab1 (col INT) USING parquet")
+        sql("INSERT INTO tab1 SELECT 1")
+        // scalastyle:off hadoopConfiguration
+        val hadoopConf = spark.sparkContext.hadoopConfiguration
```

Comment:
Can we use `spark.sessionState.newHadoopConf()? Then, we can remove `scalastyle:off hadoopConfiguration`.

Reply:
I tried to use 'spark.sessionState.newHadoopConf()' but the hadoop conf was not reflected, so ''moveToAppropriateTrash' returns false as 'fs.trash.interval' is 0

Reply:
The other places are the same. Let's try to following the Scalastyle warning if possible.

Reply:
Oh, should we change this underlying one?